### PR TITLE
`cc.CreateFromDirectFundingObjective` => `dfo.CreateConsensusChannel`

### PR DESCRIPTION
**Pull request by geoknee**
See https://github.com/statechannels/go-nitro/pull/437#discussion_r840436118 and https://github.com/statechannels/go-nitro/pull/437#discussion_r840460237

### CODE REVIEW

File: channel/consensus_channel/consensus_channel.go [hunk 1]
```diff
@@ -9,16 +9,15 @@ import (
...
 	"github.com/statechannels/go-nitro/types"
 )
 
 type ledgerIndex uint
 
 const (
-	leader   ledgerIndex = 0
-	follower ledgerIndex = 1
+	Leader   ledgerIndex = 0
+	Follower ledgerIndex = 1
```
Comment (NiloCK) [inline channel/consensus_channel/consensus_channel.go:20]: Maybe:

```suggestion
	LeaderIndex   ledgerIndex = 0
	FollowerIndex ledgerIndex = 1
```

adding more context, after exporting.

File: client/engine/engine.go [hunk 2]
```diff
@@ -267,8 +266,9 @@ func (e *Engine) attemptProgress(objective protocols.Objective) (outgoing Object
 		outgoing.CompletedObjectives = append(outgoing.CompletedObjectives, crankedObjective)
 
 		// Whenever a direct funding objective completes we want to create a consensus_channel
+		// Here we assume that every directfund.Objective is for a ledger channel.
```
Comment (NiloCK) [inline client/engine/engine.go:269]: Is it appropriate at this point to rename `directfund` to something like `createledger`?

A more generic `directfund` would probably be a from-scratch effort anyhow.

**Review by NiloCK — APPROVED** [review]
:+1:

File: channel/consensus_channel/consensus_channel.go (hunk 1 above)
Comment (geoknee) [reply channel/consensus_channel/consensus_channel.go:20]: This would add more context for consuming packages.

**Pull request merged**